src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp

```diff
@@ -0,0 +1,241 @@
+        inverse_spatial_metric,
+    const tnsr::ii<DataType, VolumeDim, Frame::Inertial>& extrinsic_curvature,
+    const tnsr::aa<DataType, VolumeDim, Frame::Inertial>& spacetime_metric,
+    const tnsr::AA<DataType, VolumeDim, Frame::Inertial>&
+        inverse_spacetime_metric,
+    const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>&
+        three_index_constraint,
+    const tnsr::aa<DataType, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_psi,
+    const tnsr::aa<DataType, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_minus,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>&
+        constraint_char_zero_plus,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>&
+        constraint_char_zero_minus,
+    const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>& phi,
+    const tnsr::ijaa<DataType, VolumeDim, Frame::Inertial>& d_phi,
+    const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>& d_pi,
+    const std::array<DataType, 4>& char_speeds) noexcept;
+/// @}
+
+namespace detail {
+template <size_t VolumeDim, typename DataType>
+void add_gauge_sommerfeld_terms_to_dt_v_minus(
+    const gsl::not_null<tnsr::aa<DataType, VolumeDim, Frame::Inertial>*>
+        bc_dt_v_minus,
+    const Scalar<DataType>& gamma2,
+    const tnsr::I<DataType, VolumeDim, Frame::Inertial>& inertial_coords,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>& incoming_null_one_form,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>& outgoing_null_one_form,
+    const tnsr::A<DataType, VolumeDim, Frame::Inertial>& incoming_null_vector,
+    const tnsr::A<DataType, VolumeDim, Frame::Inertial>& outgoing_null_vector,
+    const tnsr::Ab<DataType, VolumeDim, Frame::Inertial>& projection_Ab,
+    const tnsr::aa<DataType, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_psi) noexcept;
+
+template <size_t VolumeDim, typename DataType>
+void add_constraint_dependent_terms_to_dt_v_minus(
+    const gsl::not_null<tnsr::aa<DataType, VolumeDim, Frame::Inertial>*>
+        bc_dt_v_minus,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>& outgoing_null_one_form,
+    const tnsr::A<DataType, VolumeDim, Frame::Inertial>& incoming_null_vector,
+    const tnsr::A<DataType, VolumeDim, Frame::Inertial>& outgoing_null_vector,
+    const tnsr::aa<DataType, VolumeDim, Frame::Inertial>& projection_ab,
+    const tnsr::Ab<DataType, VolumeDim, Frame::Inertial>& projection_Ab,
+    const tnsr::AA<DataType, VolumeDim, Frame::Inertial>& projection_AB,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>&
+        constraint_char_zero_plus,
+    const tnsr::a<DataType, VolumeDim, Frame::Inertial>&
+        constraint_char_zero_minus,
+    const tnsr::aa<DataType, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_minus,
+    const std::array<DataType, 4>& char_speeds) noexcept;
+
+template <size_t VolumeDim, typename DataType>
+void add_physical_dof_terms_to_dt_v_minus(
```

Comment:
what do you think of removing `dof` from the name? I had to think about what that means, whereas `add_physical_terms` is clearer to me

Reply:
renamed

Reply:
renamed